src/protocols/secure_channel/CASESession.cpp

```diff
@@ -2179,4 +2208,33 @@ System::Clock::Timeout CASESession::ComputeSigma2ResponseTimeout(const ReliableM
         kExpectedHighProcessingTime;
 }
 
+bool CASESession::InvokeBackgroundWorkWatchdog()
+{
+    bool status = false;
+
+    if (mSendSigma3Helper && mSendSigma3Helper->UnableToScheduleAfterWorkCallback())
+    {
+        ChipLogError(SecureChannel, "SendSigma3Helper was unable to schedule the AfterWorkCallback");
+        if (mSendSigma3Helper->DoAfterWork() != CHIP_NO_ERROR)
+        {
+            SendStatusReport(mExchangeCtxt, kProtocolCodeInvalidParam);
+            mState = State::kInitialized;
```

Comment:
Should not be needed; the after-work callback should be handling all this.

Again, the DoAfterWork thing should just return void.